modules/gapi/src/backends/sycl/gsyclbackend.cpp

```diff
@@ -87,7 +87,9 @@ cv::gimpl::GSYCLExecutable::GSYCLExecutable(const ade::Graph &g,
         if (desc.storage == Data::Storage::INTERNAL && desc.shape == GShape::GMAT)
         {
           const auto mat_desc = util::get<cv::GMatDesc>(desc.meta);
-          //auto& mat = m_res.slot<cv::Mat>()[desc.rc];
+          auto& rmat = m_res.slot<cv::RMat>()[desc.rc];
+          // FIXME: Need to figure out correct RMat construction here
+          rmat = cv::make_rmat<cv::gimpl::RMatSYCLBufferAdapter>(mat holding sycl buffer?);
```

Comment:
You'll have to specify the template arguments to your adapter, and the arguments are the arguments needed by the adapter constructor.

```
rmat = cv::make_rmat<cv::gimpl::RMatSYCLBufferAdapter<float, ...>(syclBuffer);
```

Reply:
I think that I need to implement a mapping from a GMat Descriptor back to the necessary arguments for a SYCL buffer so that I can initialize one and place it into the RMat, along with the necessary template arguments.